lib/Swagger/src/ObjectSerializer.php

```diff
@@ -216,20 +220,20 @@ public function serializeCollection(array $collection, $collectionFormat, $allow
     /**
      * Deserialize a JSON string into an object
      *
-     * @param mixed  $data          object or primitive to be deserialized
-     * @param string $class         class name is passed as a string
-     * @param string $httpHeaders   HTTP headers
-     * @param string $discriminator discriminator if polymorphism is used
+     * @param mixed    $data          object or primitive to be deserialized
+     * @param string   $class         class name is passed as a string
+     * @param string[] $httpHeaders   HTTP headers
+     * @param string   $discriminator discriminator if polymorphism is used
      *
-     * @return object an instance of $class
+     * @return object|array|null an single or an array of $class instances
      */
     public static function deserialize($data, $class, $httpHeaders = null, $discriminator = null)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this function "deserialize" is 26 which is greater than 20 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of returns of this function 9, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)